src/index.ts

```diff
@@ -118,15 +121,18 @@ export default class SwupScrollPlugin extends Plugin {
 		}
 
 		// scroll to the top of the page when a visit starts, before replacing the content
-		this.on('visit:start', this.onVisitStart);
+		this.before('visit:start', this.onBeforeVisitStart, { priority: -1 });
+		this.on('visit:start', this.onVisitStart, { priority: 1 });
```

Comment:
Split visit start into two phases: one to set the flags in the visit object, one to execute the scrolling after allowing plugins and users to hook and manipulate the visit object.